AutoCollection/NativePerformance.ts

```diff
@@ -51,6 +54,10 @@ export class AutoCollectNativePerformance {
                 const NativeMetricsEmitters = require("applicationinsights-native-metrics");
                 AutoCollectNativePerformance._emitter = new NativeMetricsEmitters();
                 AutoCollectNativePerformance._metricsAvailable = true;
+                // Should only set statsbeat feature if it's the first time calling enabled
+                if (!this._isEnabled) {
```

Comment:
isn't isEnabled the config that says if we should collectNative metrics or not?  If you want to send stats only after its enabled in config and if its the first time we are initializing it, it should be checking isInitialized seems like the right thing to do.  That said, AFAICT there is place where we are setting isInitialized=true after initialization!

Reply:
The problem begins where if a user does something like enabling `NativePerformance`, thus setting initialized and enabled to true, then `dispose()` (which only sets enabled to false, but does not set initialized to false) and then calls enable again we end up in a state where initialized is true, but `this._isEnabled` is false. In that case we would want to add the native metrics feature to statsbeat, even though initialization has already happened.

Reply:
I agree that the logic above is confusing though since it's counter intuitive to see a check for `!this._isEnabled` setting the feature on statsbeat. Let me take a look at this logic and see if I can make it a bit easier to read.